File: niupanel-sdk/src/node/test_sdk.js
const sdk = require('./niu/index');

if (!process.env.NiuPanel_Key) {
    console.warn("WARNING: NiuPanel_Key environment variable is not set.");
}

async function runTests() {
    console.log("=== NiuPanel Node.js SDK Test Suite ===\n");

    // 1. 系统概览
    try {
        const overview = await sdk.getOverview();
        console.log("[PASS] Overview retrieved.");
    } catch (e) {
        console.error("[FAIL] Overview:", e.message);
    }

    // 2. 变量操作 (新流程)
    console.log("\n[Testing Variables]");
    const testKey = `test_node_${Date.now()}`;
    const testVal = "init_value";

    try {
        // 创建
        console.log(`    Creating: ${testKey}`);
        await sdk.createVariable({
            key: testKey,
            value: testVal,
            scope: "Global",
            remarks: "SDK Test"
        });

        // 获取全部模型
        const vars = await sdk.getVariable(testKey);
        console.log(`    Models found: ${vars.length}`);
        if (vars.length > 0) {
            const varId = vars[0].id;
            console.log(`    Target ID: ${varId}`);

            // 获取全部值
            const values = await sdk.getVariableValues(testKey);
            console.log(`    Values list: ${JSON.stringify(values)}`);

            // 精准更新
            console.log(`    Updating ID ${varId}...`);
            await sdk.updateVariable(varId, {
                key: testKey,
                value: "updated_value"
            });

            // 验证更新
            const updatedVal = await sdk.getVariableValues(testKey);
            console.log(`    New value: ${updatedVal[0]}`);

            // 精准删除
            console.log(`    Deleting ID ${varId}...`);
            await sdk.deleteVariable(varId);
            console.log("    [PASS] Variable CRUD");
        }
    } catch (e) {
        console.error("    [FAIL] Variables:", e.message);
    }

    // 3. 任务与文件
    try {
        const tasks = await sdk.listTasks();
        console.log(`\n[PASS] Tasks list retrieved (${tasks.items ? tasks.items.length : 0} items)`)

        const files = await sdk.listFiles("");
        console.log("[PASS] Files list retrieved.");
    } catch (e) {
        console.error("\n[FAIL] Tasks/Files:", e.message);
    }

    // 4. Jobs
    console.log("\n[Testing Jobs]");
    try {
        const jobs = await sdk.listJobs();
        console.log(`[PASS] Jobs list retrieved (${jobs.items ? jobs.items.length : 0} items)`);
    } catch (e) {
        console.error("[FAIL] Jobs:", e.message);
    }

    // 5. Environments
    console.log("\n[Testing Environments]");
    try {
        const versions = await sdk.listAvailableVersions();
        console.log(`[PASS] Available versions: ${JSON.stringify(versions)}`);
    } catch (e) {
        console.error("[FAIL] Environments:", e.message);
    }

    // 6. Compiler
    console.log("\n[Testing Compiler]");
    try {
        const compilerVersions = await sdk.getCompilerVersions();
        console.log(`[PASS] Compiler versions: ${JSON.stringify(compilerVersions)}`);
    } catch (e) {
        console.error("[FAIL] Compiler:", e.message);
    }

    // 7. Webhook 推送
    console.log("\n[Testing Webhook]");
    try {
        await sdk.pushNotification("Node.js SDK Test", "Hello from Node.js SDK!", "info");
        console.log("[PASS] Webhook push successful.");
    } catch (e) {
        console.error("[FAIL] Webhook push:", e.message);
    }
}

runTests();

File: niupanel-sdk/src/python/test_sdk.py
import os
import time
from niu import niu as sdk

# 设置日志级别以查看警告
import logging
logging.basicConfig(level=logging.INFO)

def run_tests():
    print("=== NiuPanel Python SDK Test Suite ===\n")

    # 1. 概览
    try:
        overview = sdk.get_overview()
        print("[PASS] Overview retrieved.")
    except Exception as e:
        print(f"[FAIL] Overview: {e}")

    # 2. 变量精准操作
    print("\n[Testing Variables]")
    test_key = f"test_py_{int(time.time())}"

    try:
        # 创建
        print(f"    Creating: {test_key}")
        sdk.create_variable({
            "key": test_key,
            "value": "initial",
            "scope": "Global",
            "remarks": "Python SDK Test"
        })

        # 获取全部数据模型
        vars_list = sdk.get_variable(test_key)
        print(f"    Models found: {len(vars_list)}")

        if vars_list:
            target = vars_list[0]
            var_id = target['id']
            print(f"    Target ID: {var_id}")

            # 获取所有值
            values = sdk.get_variable_values(test_key)
            print(f"    Values list: {values}")

            # 精准更新
            print(f"    Updating ID {var_id}...")
            sdk.update_variable(var_id, {
                "key": test_key,
                "value": "updated_by_py"
            })

            # 校验
            new_val = sdk.get_variable_values(test_key)
            print(f"    New value: {new_val[0]}")

            # 精准删除
            print(f"    Deleting ID {var_id}...")
            sdk.delete_variable(var_id)
            print("    [PASS] Variable CRUD")
        else:
            print("    [FAIL] Could not find created variable.")

    except Exception as e:
        print(f"    [FAIL] Variables: {e}")

    # 3. 任务与文件
    try:
        tasks = sdk.list_tasks()
        print(f"\n[PASS] Tasks list retrieved.")

        files = sdk.list_files("")
        print("[PASS] Files list retrieved.")
    except Exception as e:
        print(f"\n[FAIL] Tasks/Files: {e}")

    # 4. Jobs
    print("\n[Testing Jobs]")
    try:
        jobs = sdk.list_jobs()
        print(f"[PASS] Jobs list retrieved.")
    except Exception as e:
        print(f"[FAIL] Jobs: {e}")

    # 5. Environments
    print("\n[Testing Environments]")
    try:
        versions = sdk.list_available_versions()
        print(f"[PASS] Available versions: {versions}")
    except Exception as e:
        print(f"[FAIL] Environments: {e}")

    # 6. Compiler
    print("\n[Testing Compiler]")
    try:
        compiler_versions = sdk.get_compiler_versions()
        print(f"[PASS] Compiler versions: {compiler_versions}")
    except Exception as e:
        print(f"[FAIL] Compiler: {e}")

    # 7. Webhook 推送
    print("\n[Testing Webhook]")
    try:
        sdk.push_notification("Python SDK Test", "Hello from Python SDK!", "info")
        print("[PASS] Webhook push successful.")
    except Exception as e:
        print(f"[FAIL] Webhook push: {e}")

if __name__ == "__main__":
    if not os.environ.get("NiuPanel_Key"):
        print("Warning: NiuPanel_Key not set in environment.\n")
    run_tests()
